dvc/scm/__init__.py

```diff
@@ -1,13 +1,14 @@
 """Manages source control systems (e.g. Git)."""
 
-from dvc.scm.base import Base
+from dvc.scm.base import Base, NoSCMError
 from dvc.scm.git import Git
 
 
 # Syntactic sugar to signal that this is an actual implementation for a DVC
 # project under no SCM control.
 class NoSCM(Base):
-    pass
+    def __getattr__(self, name):
```

Comment:
A bit worried about this being too radical of a change, as it might be used somewhere. But no tests failed and I don't see any obvious issues right now, so let's try and see. :+1:

Reply:
This built in method could only be invoked if the attribute wasn't found the usual ways. It overrides all the exceptions like `'NoSCM' object has no attribute 'resolve_rev'` to A `NoSCMError`.